### PR TITLE
Accept extension of JSON content-type

### PR DESCRIPTION
**Pull request by lmazuel**
azure-core portage of https://github.com/Azure/msrest-for-python/pull/167 accepted by @johanste

### CODE REVIEW

File: sdk/core/azure-core/tests/test_universal_pipeline.py [hunk 1]
```diff
@@ -160,6 +160,12 @@ def body(self):
     result = response.context["deserialized_data"]
     assert result["success"] is True
 
+    # Simple JSON with complex content_type
+    response = build_response(b'{"success": true}', content_type="application/vnd.microsoft.appconfig.kv+json")
+    raw_deserializer.on_response(None, response)
+    result = response.context["deserialized_data"]
+    assert result["success"] is True
+
```
Comment (bryevdv) [inline sdk/core/azure-core/tests/test_universal_pipeline.py]: test for `"text/...+json"`?

Comment (lmazuel) [reply sdk/core/azure-core/tests/test_universal_pipeline.py]: Fixed!

File: sdk/core/azure-core/azure/core/pipeline/policies/universal.py [hunk 2]
```diff
@@ -266,10 +266,9 @@ def on_response(self, request, response):
 class ContentDecodePolicy(SansIOHTTPPolicy):
     """Policy for decoding unstreamed response content.
     """
-    JSON_MIMETYPES = [
-        'application/json',
-        'text/json' # Because we're open minded people...
-    ]
+    # Accept "text" because we're open minded people...
+    JSON_REGEXP = re.compile(r'^(application|text)/([a-z+.]+\+)?json$')
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: Do we know 100% that there will never be digits? I've definitely encountered custom mime types with e.g. `"v1"` in them in some contexts: https://jupyter.readthedocs.io/en/latest/reference/mimetype.html I am not sure how common it is in general, or if there is some basis for extra assumptions here.

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: I'm fine to add digit.

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: Fixed

**Pull request merged**